File: hs/1.hs
import System.IO
import Data.List

totalUp:: [Int] -> Int -> [Int]
totalUp arr num = 
  if num == 0 || length arr == 0
  then arr ++ [0]
  else init arr ++ [num + last arr]

repla :: String -> String
repla "" = "0"
repla str = str

main :: IO ()
main = do 
  ints:: [Int] <- foldl totalUp [] .map read . map repla . lines <$> readFile "../src/1.txt"
  let part1 = maximum ints
  let part2 = (sum . take 3 .reverse .sort)ints
  print part1
  print part2

            

File: rs/1.rs
use std::fs;

fn main() {
    let data = fs::read_to_string("../../src/1.txt").expect("To read file");

    let mut highest = data.lines().fold(Vec::new(), |mut a, b| {
        if b == "" || a.len() == 0 {
            a.push(0);
        } else {
            let c = a.last_mut().expect("Should be a number");
            *c += b.parse::<u32>().expect("Should be a number");
        }
        return a;
    });

    highest.sort();
    highest.reverse();

    println!("Part 1: {}", highest[0]);
    println!("Part 2: {}", highest.iter().take(3).sum::<u32>())
}
